### PR TITLE
HIVE-1498: Add InvalidInstallConfigSubnet install log regex.

### PR DESCRIPTION
**Pull request by abutcher**
/assign @dgoodwin 

I also added the regex to the additional template which I'm testing in integration.

### CODE REVIEW

**Comment by abutcher** [comment]
Example error we're catching:

```
msg=failed to fetch Master Machines: failed to load asset \"Install Config\": [platform.aws.subnets[0]: Invalid value: \"subnet-123456\": subnet's CIDR range start 10.1.1.0 is outside of the specified machine networks, platform.aws.subnets[1]: Invalid value: \"subnet-123456\": subnet's CIDR range start 10.1.100.0 is outside of the specified machine networks]\n"
```

File: config/configmaps/install-log-regexes-configmap.yaml [hunk 1]
```diff
@@ -31,6 +31,11 @@ data:
       - "LimitExceeded"
       installFailingReason: ResourceLimitExceeded
       installFailingMessage: Resource limit exceeded
+    - name: InvalidSubnetCIDRConfig
+      searchRegexStrings:
+      - "subnet\'s CIDR range"
+      installFailingReason: InvalidSubnetCIDRConfig
+      installFailingMessage: Invalid subnet CIDR provided in install config.
```
Comment (dgoodwin) [inline config/configmaps/install-log-regexes-configmap.yaml:38]: Slight nit, the regex looks just slightly lacking in an indication of something negative, could we extend a little further to include "is outside of the specified machine networks".

File: hack/regexes/additional-hive-regexes.yaml [hunk 2]
```diff
@@ -14,3 +14,8 @@ objects:
             - "EIP: AddressLimitExceeded"
           installFailingReason: EIPAddressLimitExceeded
           installFailingMessage: EIP Address limit exceeded
+        - name: InvalidSubnetCIDRConfig
+          searchRegexStrings:
+            - "subnet\'s CIDR range"
+          installFailingReason: InvalidSubnetCIDRConfig
+          installFailingMessage: Invalid subnet CIDR provided in install config.
```
Comment (dgoodwin) [inline hack/regexes/additional-hive-regexes.yaml:21]: What value does having in additional-hive-regexes have? Does this tie into what rolls out immediately?

**Comment by dgoodwin** [comment]
> What about generically checking for `failed to load asset \"Install Config\"` as well?

Seems reasonable for low in the list.

File: hack/regexes/additional-hive-regexes.yaml (hunk 2 above)
Comment (abutcher) [reply hack/regexes/additional-hive-regexes.yaml:21]: Yes.

File: config/configmaps/install-log-regexes-configmap.yaml (hunk 1 above)
Comment (abutcher) [reply config/configmaps/install-log-regexes-configmap.yaml]: Made more specific (assuming we want to go ahead with adding it for now) and renamed to `InvalidInstallConfigSubnet`

**Comment by dgoodwin** [comment]
Business as usual until we sort out that proposal from Abhinav.

Thanks!

/lgtm

**Pull request merged**